frontend/packages/topology/src/anchors/SVGAnchor.ts

```diff
@@ -17,6 +17,10 @@ export default class SVGAnchor extends AbstractAnchor {
   }
 
   getCircleLocation(circle: SVGCircleElement, reference: Point): Point {
+    if (!circle.viewportElement) {
```

Comment:
Can't you do this once in getLocation?

Reply:
Though they are not part of the `Anchor` interface, it is possible that someone use those functions directly and then the problem would still exist.

Reply:
Those functions should be `private`.

Reply:
Moved the check to `getLocation` made the `get<xxx>Location` functions private.